huffman/test/bit_span_test.cpp

```diff
@@ -84,12 +83,71 @@ auto main() -> int
   using ::boost::ut::operator|;
 
   test("aborts if bit offset too large") = [](auto bit_offset) {
-    static constexpr auto data =
-        std::array{std::byte{0b10101010}, std::byte{0xff}};
+    static constexpr auto data = huffman::byte_array(0b10101010, 0xff);
 
     expect(aborts([&] {
       static constexpr auto bit_size = 7;
       huffman::bit_span{data.begin(), bit_size, bit_offset};
     }));
   } | std::vector<std::uint8_t>{8, 9, 10};  // NOLINT(readability-magic-numbers)
+
+  std::vector<unsigned> n_to_consume(1Z + 2Z * CHAR_BIT);
+  std::iota(n_to_consume.begin(), n_to_consume.end(), 0);
+
+  test("consume") = [](auto n) {
+    static constexpr auto data = huffman::byte_array(0b10101010, 0b01010101);
+
+    static constexpr auto nth_bit = [](auto m) {
+      return huffman::bit{std::bitset<CHAR_BIT>{
+          std::to_integer<unsigned>(data[m / CHAR_BIT])}[m % CHAR_BIT]};
+    };
+
+    auto bits = huffman::bit_span{data};
+    if (n <= data.size() * CHAR_BIT) {
+      bits.consume(n);
+      expect(bits.n_consumed() == n);
```

Comment:
I don't like the idea of tracking history in the state of a bit_span.

For example, what is the expected value here?

```cpp
auto data = ...;

auto b1 = bit_span{data};
b1.consume(CHAR_BIT);

assert(b1.n_consumed() == CHAR_BIT);

auto b2 = b1;
assert(b2.n_consumed() == ??);
```

```suggestion
      const auto initial_bits = ...; // up above
      expect(initial_bits.size() - bits.size() == n);
```